crates/configuration/src/configuration.rs

```diff
@@ -47,31 +50,60 @@ pub struct Configuration {
 }
 
 pub async fn introspect(
-    input: RawConfiguration,
+    input: ParsedConfiguration,
     environment: impl Environment,
-) -> anyhow::Result<RawConfiguration> {
+) -> anyhow::Result<ParsedConfiguration> {
     match input {
-        RawConfiguration::Version3(config) => Ok(RawConfiguration::Version3(
+        ParsedConfiguration::Version3(config) => Ok(ParsedConfiguration::Version3(
             version3::introspect(config, environment).await?,
         )),
-        RawConfiguration::Version4(config) => Ok(RawConfiguration::Version4(
+        ParsedConfiguration::Version4(config) => Ok(ParsedConfiguration::Version4(
             version4::introspect(config, environment).await?,
         )),
     }
 }
 
 pub async fn parse_configuration(
     configuration_dir: impl AsRef<Path> + Send,
-    environment: impl Environment,
-) -> Result<Configuration, Error> {
+) -> Result<ParsedConfiguration, ParseConfigurationError> {
     // Try parsing each supported version in turn
-    match version4::parse_configuration(configuration_dir.as_ref(), &environment).await {
-        Err(v4_err) => {
-            match version3::parse_configuration(configuration_dir.as_ref(), &environment).await {
-                Err(v3_err) => Err(Error::UnableToParseAnyVersions(vec![v3_err, v4_err])),
-                Ok(config) => Ok(config),
-            }
+    match version4::parse_configuration(configuration_dir.as_ref()).await {
+        Err(v4_err) => match version3::parse_configuration(configuration_dir.as_ref()).await {
+            Err(v3_err) => Err(ParseConfigurationError::UnableToParseAnyVersions(
+                MultiError(vec![Box::new(v4_err), Box::new(v3_err)]),
+            )),
+            Ok(config) => Ok(ParsedConfiguration::Version3(config)),
+        },
+        Ok(config) => Ok(ParsedConfiguration::Version4(config)),
+    }
+}
+
+pub fn make_runtime_configuration(
```

Comment:
Could you please add a comment about what this do/what's the difference between a parsed conf and a runtime conf?

Reply:
Yeah, good idea. There are certainly some important nuances that it would serve us well to document. Some of this is in https://github.com/hasura/ndc-postgres/pull/452#discussion_r1589400118 but it ought to be in the code as well.